api/base/serializers.py

```diff
@@ -60,12 +61,16 @@ def get_url(self, obj, view_name, request, format):
         """
         Given an object, return the URL that hyperlinks to the object.
 
-        Returns null if lookup value is None
+        Returns None if lookup value is None
         """
-
-        if getattr(obj, self.lookup_field) is None:
+        if deep_get(obj, self.lookup_field) is None:
             return None
 
+        head, tail = os.path.split(self.lookup_field.replace('.', '/'))
```

Comment:
os.path.split breaks the string into two segments: 

i.e. '/home/sam/osf.io/requirements.txt' => (' '/home/sam/osf.io/', 'requirements.txt')

whereas a regular str.split produces a list.